nio.conf

```diff
@@ -12,6 +12,8 @@ PROJECT_ROOT=
 
 INSTANCE_NAME=local
 INSTANCE_ID=local
+# specifies the api key to use when making requests to the product api
+API_KEY=
```

Comment:
Does this actually load from the `api_key.cfg` file, or how would that value be populated? I really wouldn't want to have to include documentation for the user to have to go and update their `API_KEY` value manually.

Reply:
I could see us, in the future, wanting to have the designer auto-set the `INSTANCE_ID` as well, so we would want to figure out the order of precedence to accommodate that as well.

If `INSTANCE_ID / API_KEY` is blank in `nio.conf`, then load from persistence?

Reply:
fwiw, `diagnostic component` has PR (approved not merged) that follows same precedence as `deployment component`, that is, `Persistence.load` overrides `Settings.get`, furthermore, it allows for the `api_key` to come as part of the request, in which case, it is saved to Persistence